src/CMakeLists.txt

```diff
@@ -47,6 +47,7 @@ configure_file(csympy_config.h.in csympy_config.h)
 include_directories(BEFORE ${csympy_BINARY_DIR}/src)
 
 add_library(csympy ${SRC})
+target_link_libraries(csympy teuchos ${LIBS})
```

Comment:
@isuruf is my understanding correct, that this line causes the following:

a) if `csympy` is a static lib, then cmake will add `teuchos` and `${LIBS}` into gcc options for each linking of `csympy`
b) if `csympy` is a dynamic lib, then `teuchos` (which is a static lib) will get included in it, as well as any static libs from `${LIBS}` and the rest (i.e. the dynamic libs) will be added to gcc link options.